spinoso-time/src/time/chrono/build.rs

```diff
@@ -50,11 +53,56 @@ impl Time {
             offset,
         }
     }
+
+    /// Creates a new `Time` object from the `seconds` and `sub_seconds_nanos`
+    /// since the Unix EPOCH with a local offset.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::Time;
+    /// let epoch = Time::at(0, 0);
+    /// let epoch_plus_1_nano = Time::at(0, 1);
+    /// ```
+    #[inline]
+    #[must_use]
+    pub fn at(seconds: i64, sub_second_nanos: i64) -> Option<Self> {
+        let offset = Offset::Local;
+
+        let overflow = sub_second_nanos / i64::from(NANOS_IN_SECOND);
+        let timestamp = seconds.checked_add(overflow)?;
+        let sub_second_nanos = sub_second_nanos % i64::from(NANOS_IN_SECOND);
+
+        // Sub-seconds are stored as a non-negative. So negative sub-seconds are
+        // handled by subtracting one full second and calculating a new sub-second value:
+        //
+        // ```console
+        // [2.6.3] > Time.at(0, -1).to_i
+        // => -1
+        // [2.6.3] > Time.at(0, -1).nsec
+        // => 999999000
+        // ```
+        let (timestamp, sub_second_nanos) = if sub_second_nanos > 0 {
+            (timestamp, sub_second_nanos)
+        } else {
+            (
+                timestamp.checked_sub(1)?,
+                i64::from(NANOS_IN_SECOND).checked_sub(sub_second_nanos.abs())?,
+            )
+        };
+
+        Some(Self {
+            timestamp,
+            sub_second_nanos: u32::try_from(sub_second_nanos).ok()?,
```

Comment:
nit: I'd like to see the type of `sub_second_nanos` here to already be a `u32` so the construction of the struct doesn't have any complex logic in it.